libcloud/dns/drivers/godaddy.py

```diff
@@ -0,0 +1,521 @@
+
+    allow_insecure = False
+
+    def __init__(self, key, secret, shopper_id, secure=True, host=None,
+                 port=None, url=None, timeout=None,
+                 proxy_url=None, backoff=None, retry_delay=None):
+        super(GoDaddyDNSConnection, self).__init__(
+            key,
+            secure=secure, host=host,
+            port=port, url=url,
+            timeout=timeout,
+            proxy_url=proxy_url,
+            backoff=backoff,
+            retry_delay=retry_delay)
+        self.key = key
+        self.secret = secret
+        self.shopper_id = shopper_id
+
+    def add_default_headers(self, headers):
+        headers['X-Shopper-Id'] = self.shopper_id
+        headers['Authorization'] = "sso-key %s:%s" % \
+            (self.key, self.secret)
+        return headers
+
+
+class GoDaddyDNSDriver(DNSDriver):
+    """
+    A driver for GoDaddy DNS.
+
+    This is for customers of GoDaddy
+    who wish to purchase, update existing domains
+    and manage records for DNS zones owned by GoDaddy NS servers.
+    """
+    type = Provider.GODADDY
+    name = 'GoDaddy DNS'
+    website = 'https://www.godaddy.com/'
+    connectionCls = GoDaddyDNSConnection
+
+    RECORD_TYPE_MAP = {
+        RecordType.A: 'A',
+        RecordType.AAAA: 'AAAA',
+        RecordType.CNAME: 'CNAME',
+        RecordType.MX: 'MX',
+        RecordType.NS: 'SPF',
+        RecordType.SRV: 'SRV',
+        RecordType.TXT: 'TXT',
+    }
+
+    def __init__(self, shopper_id, key, secret,
+                 secure=True, host=None, port=None):
+        """
+        Instantiate a new `GoDaddyDNSDriver`
+
+        :param  shopper_id: Your customer ID or shopped ID with GoDaddy
+        :type   shopper_id: `str`
+
+        :param  key: Your access key from developer.godaddy.com
+        :type   key: `str`
+
+        :param  secret: Your access key secret
+        :type   secret: `str`
+        """
+        super(GoDaddyDNSDriver, self).__init__(key=key, secret=secret,
+                                               secure=secure,
+                                               host=host, port=port,
+                                               shopper_id=str(shopper_id))
+
+    def _to_zones(self, items):
+        zones = []
+        for item in items:
+            zones.append(self._to_zone(item))
+        return zones
+
+    def _to_zone(self, item):
+        extra = {"expires": item['expires']}
+        zone = Zone(id=item['domainId'], domain=item['domain'],
+                    type='master', ttl=None,
+                    driver=self, extra=extra)
+        return zone
+
+    def _to_records(self, items, zone=None):
+        records = []
+
+        for item in items:
+            records.append(self._to_record(item=item, zone=zone))
+        return records
+
+    def _to_record(self, item, zone=None):
+        ttl = item['ttl']
+        type = self._string_to_record_type(item['type'])
+        name = item['name']
+        id = '{0}:{1}'.format(name, type)
```

Comment:
Please use '%s' for string interpolation. We still support Python 2.5 and 2.6 (we will probably drop support for 2.5 in the near future) where `.format()` is not available.